### DISCUSSION TITLE
Is there any way to restrict some vehicles to go from Node A to Node B

### DISCUSSION THREAD

**Discussion by kazitanvirahsan** [google/or-tools#2841] [Routing (and legacy CP) questions] [answered]
Hello,

I want to restrict some vehicles to travel for example from node A to node B. Other Vehicles can travel
from Node A to Node B. 

What I tried so far is routing.NextVar(pickup_index_1).RemoveValue(pickup_index_2) but this is route-specific not vehicle specific. Is there any way that I can able to assign paths to vehicles and vehicles can only travel to those paths?

For example,
For Vehicle A     can go  A->B, A->C, C->D 
       Vehicle B     can go A->B, A->C  (can not go C->D)

One way or another. Any idea of how to approach this problem?

Kazi

**Comment by jmarca** [comment]
Use a conditional constraint, where the condition is vehicle is in the set you want to restrict. 

Before you reply asking for an example, for sure I’ve posted examples of conditional constraints to this forum before, so please search. 

Also, they are hard to use and documentation is thin, so you’ll have to experiment a bit to figure out the semantics. 

James

**Comment by lperron** [comment, marked as answer]
You can also have different distance matrix per vehicles. And for large trucs, have arcs that are too expensive.

**Reply by kazitanvirahsan** [reply]
@lperron - Thank you so much! It really helps.

**Reply by jmarca** [reply]
Yeah that’s a much better idea than mine! And the funny thing is, I use that approach all the time. Not enough coffee I guess
